godot-core/src/task/futures.rs

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use core::panic;
+use std::fmt::Display;
+use std::future::{Future, IntoFuture};
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll, Waker};
+
+use crate::builtin::{Callable, RustCallable, Signal, Variant};
+use crate::classes::object::ConnectFlags;
+use crate::meta::ParamTuple;
+use crate::obj::{EngineBitfield, WithBaseField};
+use crate::registry::signal::TypedSignal;
+
+/// The panicking counter part to the [`FallibleSignalFuture`].
+///
+/// This future works in the same ways as the `FallibleSignalFuture`, but panics when the signal object is freed instead of resolving to an
+/// [`Result::Err`]
```

Comment:
```suggestion
/// The panicking counter part to the [`FallibleSignalFuture`].
///
/// This future works in the same way as `FallibleSignalFuture`, but panics when the signal object is freed, instead of resolving to a
/// [`Result::Err`].
```